docs/content/tutorial/managing-your-own-io.mdx

```diff
@@ -0,0 +1,106 @@
+---
+title: "Tutorial, part seven: Managing your own I/O | Dagster Docs"
+description: Learn how to use I/O managers to save your data.
+---
+
+# Tutorial, part 7: Managing your own I/O
+
+When building the dictionary or word counts, the `most_frequent_words` asset currently uses a static list of stopwords to decide what words to ignore. You’ll notice that despite the length of the list, many unhelpful words and prepositions are still not filtered out. You’d like to ignore those words. To scale this properly, you’ll use a CSV of stopwords found on the internet, use it for `most_frequent_words`, and update it before the refresh to ensure that your list of stopwords is up-to-date.
+
+While doing so, you'll take a step back and learn how to use more of Dagster's versatility without using every feature provided to you. Throughout this tutorial, you've learned multiple Dagster concepts that explain how Dagster can manage functionality on your behalf, ex. I/O managers to read and write from storage.
+
+You'll create assets to manage reading and writing from storage on your own. This is useful for multiple purposes, such as:
+
+- Your pipeline manages I/O on its own
+- You want to run a SQL query that creates or updates a table on the database.
+- Adjacent to the above point, your data is too large to fit into memory.
+- You don't want to use I/O managers.
+
+By the end of this section, you'll also have assets that depend on other assets without loading the dependent asset's data into memory.
+
+## Step 1: Creating an asset without an I/O Manager
+
+Many valid situations exist where you would want to avoid using an I/O manager to write an asset back to memory. One is when the asset is already saved in the asset function definition without having to return the data.
```

Comment:
"One is when the asset is already saved in the asset function definition without having to return the data." - this feels a bit circular - i.e. we're not using an I/O manager because we're not using an I/O manager.

I would say something like "we're invoking an API that writes data to disk, rather than returning it to us in memory. Therefore, we don't need an I/O manager to handle writing the data to disk."